apps/src/code-studio/videos.js

```diff
@@ -287,14 +266,72 @@ videos.showVideoDialog = function(options, forceShowVideo) {
   // Don't add fallback player if a video modal has closed
   var shouldStillAdd = true;
   videoModal.one('hidden.bs.modal', function() {
+    window.removeEventListener('resize', resizeVideoPlayerListener);
+    window.removeEventListener('scroll', resizeVideoPlayerListener);
     shouldStillAdd = false;
   });
 
+  var divHeight = getVideoHeight();
   setupVideoFallback(options, $div.width(), divHeight, function() {
     return shouldStillAdd;
   });
 };
 
+/**
+ * Sets the video player container to 80% of the width or height of the screen
+ * at a 9/16 ratio.
+ */
+function resizeVideoPlayer() {
+  let $div = $(MODAL_CLASS);
+  const navBarHeight = $div.find(TAB_NAV_CLASS).outerHeight();
+  const screenRatio = 0.8;
+  const aspectRatio = 16 / 9;
+  const maxHeight = window.innerHeight * screenRatio;
+  const maxWidth = window.innerWidth * screenRatio;
+
+  let containerHeight = maxWidth / aspectRatio + navBarHeight;
+  let containerWidth = maxWidth;
+  if ((maxHeight - navBarHeight) * aspectRatio < maxWidth) {
+    containerHeight = maxHeight;
+    containerWidth = (maxHeight - navBarHeight) * aspectRatio;
+  }
```

Comment:
It looks like this is where either maxWidth or maxHeight becomes the dominant value?  It would be nice to do this with immutable values, so that we don't set containerWidth and then sometimes overwrite it immediately afterwards, instead clearly only setting it once based on the winning condition.  Maybe a brief comment would be good too.